### PR TITLE
Added ability to set Status Description & Request/Response Schema name in ResourceSnippetsParameters

### PR DESCRIPTION
**Pull request by oleksandr-abasov**

### CODE REVIEW

**Comment by oleksandra-lohika-tix** [comment]
Hi team,
here is the PR to add extra params to ResourceSnippetParameters to set 

- status description
- request schema name
- response schema name

Here is the usage example:
```
.andDo(document("pet-create",
                        preprocessResponse(prettyPrint()),
                        snippet(
                                ResourceSnippetParameters(
                                        tags = PET_TAGS,
                                        description = "Create pet",
                                        statusDescription = "Pet was successfully created",
                                        requestSchemaName = "pet_create_request",
                                        responseSchemaName = "pet_create_response",
                                        requestFields = petRequestFields,
                                        responseFields = petResponseFields
                                )
                        )
                ))
```

The fields are optional, so when you haven't provided params the old logic of each one will work as before.

**Comment by mduesterhoeft** [comment]
Thanks for your PR 🙏. I am off for a few days. So my review might take a few days.

**Comment by thowimmer** [comment]
Hi all,

I just started the implementation of the identical feature due to issue #78.
Cool to see that some other contributors already invested in this feature :-)

I was wondering whether it would be a more future proof approach if schema specific properties, such as the name of the schema, are put into a complex object:

```java
.andDo(document("pet-create",
                        preprocessResponse(prettyPrint()),
                        snippet(
                                ResourceSnippetParameters(
                                        tags = PET_TAGS,
                                        description = "Create pet",
                                        statusDescription = "Pet was successfully created",
                                        requestSchema(name = "pet_create_request"),
                                        responseSchema(name = "pet_create_response"),
                                        requestFields = petRequestFields,
                                        responseFields = petResponseFields
                                )
                        )
                ))
```

Feature wise this is no difference, but it enables to expand the schema properties in the future - Thinking of [inheritance and polymorphism](https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/) support etc.
Moreover such a structure could be used for field level schemas for describing and [reusing nested object structures](https://swagger.io/docs/specification/using-ref/).

File: restdocs-api-spec/src/main/kotlin/com/epages/restdocs/apispec/ResourceSnippet.kt [hunk 1]
```diff
@@ -69,14 +69,17 @@ class ResourceSnippet(private val resourceSnippetParameters: ResourceSnippetPara
                 headers = resourceSnippetParameters.requestHeaders.withExampleValues(operation.request.headers),
                 pathParameters = resourceSnippetParameters.pathParameters.filter { !it.isIgnored },
                 requestParameters = resourceSnippetParameters.requestParameters.filter { !it.isIgnored },
+                schema = resourceSnippetParameters.requestSchemaName,
```
Comment (mduesterhoeft) [inline restdocs-api-spec/src/main/kotlin/com/epages/restdocs/apispec/ResourceSnippet.kt]: Let's also call this `schemaName` here

File: restdocs-api-spec-openapi3-generator/src/main/kotlin/com/epages/restdocs/apispec/openapi3/OpenApi3Generator.kt [hunk 2]
```diff
@@ -299,7 +300,7 @@ object OpenApi3Generator {
             .groupBy { it.response.contentType!! }
 
         val apiResponse = ApiResponse().apply {
-            description = responseModelsSameStatus.first().response.status.toString()
+            description = responseModelsSameStatus.first().response.statusDescription ?: responseModelsSameStatus.first().response.status.toString()
```
Comment (mduesterhoeft) [inline restdocs-api-spec-openapi3-generator/src/main/kotlin/com/epages/restdocs/apispec/openapi3/OpenApi3Generator.kt]: 👍

File: restdocs-api-spec/src/main/kotlin/com/epages/restdocs/apispec/ResourceSnippet.kt [hunk 3]
```diff
@@ -127,6 +130,7 @@ class ResourceSnippet(private val resourceSnippetParameters: ResourceSnippetPara
         val path: String,
         val method: String,
         val contentType: String?,
+        val schema: String? = null,
```
Comment (mduesterhoeft) [inline restdocs-api-spec/src/main/kotlin/com/epages/restdocs/apispec/ResourceSnippet.kt]: Let's also call this `schemaName` here

**Comment by mduesterhoeft** [comment]
@thowimmer thanks for pointing this out. I am not sure if we would really need the Schema object here. @oleksandra-lohika-tix what do you think?

**Comment by oleksandra-lohika-tix** [comment]
> @thowimmer thanks for pointing this out. I am not sure if we would really need the Schema object here. @oleksandra-lohika-tix what do you think?

@mduesterhoeft Thank you for the review, 
Sure, OpenApi20Generator can be changed too
I will try to fix the mentioned above and update the PR.

**Comment by mduesterhoeft** [comment]
@oleksandra-lohika-tix any news on this one?

**Comment by thowimmer** [comment]
I can also support or take over if you like :-)
Just let me know.

**Comment by oleksandra-lohika-tix** [comment]
Hi guys, sorry but have a lack of time, can't complete, tough deadlines on project
Yes, sure you can takeover from this PR

**Comment by thowimmer** [comment]
Sure I will take over. I will try to finish the remaining work in the next week.
However - I guess I have to continue on my own fork and sync the changes of @oleksandra-lohika-tix.
I will create another PR once I'm finished on my fork.

**Comment by mduesterhoeft** [comment]
Thanks @thowimmer. Looking forward to your PR

**Pull request closed without merging**